conan/api/subapi/remotes.py

```diff
@@ -35,6 +35,9 @@ def __init__(self, conan_api):
         # Wraps an http_requester to inject proxies, certs, etc
         self._requester = ConanRequester(self.conan_api.config.global_conf, self.conan_api.cache_folder)
 
+    def reinit(self):
+        self._requester = ConanRequester(self.conan_api.config.global_conf, self.conan_api.cache_folder)
```

Comment:
This explicitly recreates the object so that the cache gets invalidated - a cleaner reinit() might be nicer, but for now I think this is ok

Reply:
A minor issue remains here: `Remote` objects have a cache that saves package results, but the object that we gave the user in a `get` call is not retriveable again, so we can't invalidate them! Maybe this is a separate concern althogether, but thought I'd at least mention it

Reply:
Indeed, it is impossible to invalidate the ``Remote`` object cache, unless every ``RemoteAPI`` call that returns remotes keep a reference to them to do so, but at the moment sounds too much, probably just documenting properly the ``Remote`` behavior would be ok.